src/SQLite/Export.php

```diff
@@ -0,0 +1,282 @@
+<?php
+namespace WP_CLI\DB\SQLite;
+
+use Exception;
+use PDO;
+use WP_CLI;
+use WP_SQLite_Translator;
+
+class Export extends Base {
+
+	/**
+	 * List of arguments that are not supported by the export command.
+	 * @var string[]
+	 */
+	protected $unsupported_arguments = [
+		'fields',
+		'include-tablespaces',
+		'defaults',
+		'dbuser',
+		'dbpass',
+	];
+
+	protected $translator;
+	protected $args      = array();
+	protected $is_stdout = false;
+
+	public function __construct() {
+		$this->load_dependencies();
+		$this->translator = new WP_SQLite_Translator();
+	}
+
+	/**
+	 * Run the export command.
+	 *
+	 * @param string $result_file The file to write the exported data to.
+	 * @param array  $args        The arguments passed to the command.
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function run( $result_file, $args ) {
+		$this->args = $args;
+		$this->check_arguments( $args );
+
+		$handle = $this->open_output_stream( $result_file );
+
+		$this->write_sql_statements( $handle );
+		$this->close_output_stream( $handle );
+
+		$this->display_result_message( $result_file );
+	}
+
+	/**
+	 * Get output stream for the export.
+	 *
+	 * @param $result_file
+	 *
+	 * @return false|resource
+	 * @throws WP_CLI\ExitException
+	 */
+	protected function open_output_stream( $result_file ) {
+		$this->is_stdout = '-' === $result_file;
+		$handle          = $this->is_stdout ? fopen( 'php://stdout', 'w' ) : fopen( $result_file, 'w' );
+		if ( ! $handle ) {
+			WP_CLI::error( "Unable to open file: $result_file" );
+		}
+		return $handle;
+	}
+
+	/**
+	 * Close the output stream.
+	 *
+	 * @param $handle
+	 *
+	 * @return void
+	 * @throws WP_CLI\ExitException
+	 */
+	protected function close_output_stream( $handle ) {
+		if ( ! fclose( $handle ) ) {
+			WP_CLI::error( 'Error closing output stream.' );
+		}
+	}
+
+	/**
+	 * Write SQL statements to the output stream.
+	 *
+	 * @param resource $handle The output stream.
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	protected function write_sql_statements( $handle ) {
+		$include_tables = $this->get_include_tables();
+		$exclude_tables = $this->get_exclude_tables();
+		foreach ( $this->translator->query( 'SHOW TABLES' ) as $table ) {
+			// Skip tables that are not in the include_tables list if the list is defined
+			if ( ! empty( $include_tables ) && ! in_array( $table->name, $include_tables, true ) ) {
+				continue;
+			}
+
+			// Skip tables that are in the exclude_tables list
+			if ( in_array( $table->name, $exclude_tables, true ) ) {
+				continue;
+			}
+
+			$this->write_create_table_statement( $handle, $table->name );
+			$this->write_insert_statements( $handle, $table->name );
+		}
+
+		fwrite( $handle, sprintf( '-- Dump completed on %s', gmdate( 'c' ) ) );
```

Comment:
`fwrite()` enforces using PHP resource handles. This is a good start, and building on top of that it would be lovely to output that as a simple string for the API consumer to stream-write wherever it needs to. This would play well with [the Streaming API I'm exploring for WordPress](https://github.com/adamziel/wxr-normalize/pull/1).